File: backend/src/main/java/com/nexaerp/account/Account.java
package com.nexaerp.account;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Table(name= "accounts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String code;

    @Column(nullable = false)
    private String name;

    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AccountType type;

    @Column(nullable = false)
    private Boolean isActive = true;

    @Column(nullable = false)
    private Boolean isDefault = false;

    // Self referencing — Parent Account
    @ManyToOne
    @JoinColumn(name = "parent_id")
    private Account parent;

    // Self referencing — Child Accounts
    @OneToMany(mappedBy = "parent", cascade = CascadeType.ALL)
    private List<Account> children;

}

File: frontend/src/app/features/account/chart-of-accounts/models/account.ts
// features/account/chart-of-accounts/models/account.ts

export interface Account {
  id: number;
  code: string;
  name: string;
  description?: string;
  type: AccountType;
  isActive: boolean;
  isDefault: boolean;
  parentId?: number;
  parentName?: string;
  currentBalance: number;
  children?: Account[];
}

export type AccountType =
  'ASSET' | 'LIABILITY' | 'EQUITY' | 'REVENUE' | 'EXPENSE';

export interface AccountRequest {
  code: string;
  name: string;
  description?: string;
  type: AccountType;
  parentId?: number;
}

export interface AccountFilter {
  search?: string;
  type?: AccountType | '';
  isActive?: boolean | '';
  parentId?: number | '';
}